pubsub/gcppubsub/gcppubsub_test.go

```diff
@@ -58,9 +58,22 @@ func newHarness(ctx context.Context, t *testing.T) (drivertest.Harness, error) {
 }
 
 func (h *harness) CreateTopic(ctx context.Context, testName string) (dt driver.Topic, cleanup func(), err error) {
-	topicName := fmt.Sprintf("%s-topic-%d", sanitize(testName), atomic.AddUint32(&h.numTopics, 1))
-	topicPath := fmt.Sprintf("projects/%s/topics/%s", projectID, topicName)
-	return createTopic(ctx, h.pubClient, topicName, topicPath)
+	// We may encounter topics that were created by a previous test run and were
+	// not properly cleaned up. In such a case delete the existing token and create
+	// a new topic with a higher topic number (to avoid cool-off issues between
+	// deletion and re-creation of the same topic).
+	for {
+		topicName := fmt.Sprintf("%s-topic-%d", sanitize(testName), atomic.AddUint32(&h.numTopics, 1))
+		topicPath := fmt.Sprintf("projects/%s/topics/%s", projectID, topicName)
+		dt, cleanup, err := createTopic(ctx, h.pubClient, topicName, topicPath)
+		// TODO: not for submit; how to test this error correctly?
+		if err != nil && strings.Contains(err.Error(), "AlreadyExists") {
```

Comment:
The problem might be that we're wrapping the error inside `createTopic` below rather than returning exactly what `pubClient.CreateTopic` returns. Try removing that `fmt.Printf` and seeing if the error code thing @jba suggested works.

Reply:
Nice spot! It works now with @jba's check

Just removing the `fmt.Errorf` wrapping for now. Is this something xerrors could help with?

Reply:
Yes, in the new error world you can add text to an error but still preserve the error value itself.